packages/story-editor/src/components/form/stackedInputs/index.js

```diff
@@ -0,0 +1,193 @@
+/*
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import styled, { css } from 'styled-components';
+import { NumericInput } from '@web-stories-wp/design-system';
+/**
+ * Internal dependencies
+ */
+import { inputContainerStyleOverride } from '../../panels/shared';
+import { MULTIPLE_DISPLAY_VALUE, MULTIPLE_VALUE } from '../../../constants';
+import { getStackedInputValues } from './utils';
+
+const InputContainer = styled.div`
+  margin-left: 1px;
+  ${({ isSmall }) => (!isSmall ? 'max-width: 106px' : '')}
+`;
+
+const BoxedNumericInput = styled(NumericInput)`
+  border-radius: 0px;
+  margin-left: -1px;
+  svg {
+    ${({ side }) =>
+      side === 'topLeft' || side === 'top' ? 'transform: rotate(90deg);' : ''}
+  }
+  ${({ isSmall, side }) =>
+    isSmall &&
+    `
+   width: 25%;
+   svg {
+     ${
+       side === 'topRight' || side === 'right'
+         ? 'transform: rotate(180deg);'
+         : ''
+     }
+     ${
+       side === 'bottomRight' || side === 'bottom'
+         ? 'transform: rotate(270deg);'
+         : ''
+     }
+     width: 29px;
+     height: 29px;
+     margin-right: -14px;
+   }
+ `}
+`;
+
+const inputContainerStyle = css`
+  ${inputContainerStyleOverride}
+  position: relative;
+  :focus-within {
+    z-index: 1;
+  }
+`;
+
+const inputStyleOverride = css`
+  height: 36px;
+  padding: 2px 6px 2px 12px;
+`;
+
+const lastInputStyleOverride = css`
+  ${inputStyleOverride}
+  padding-right: 10px;
+`;
+
+const styleOverrideTopLeft = css`
+  ${inputContainerStyle}
+  border-radius: 4px 0 0 4px;
+`;
+
+const styleOverrideTopRight = css`
+  ${inputContainerStyle}
+  border-radius: 0;
+`;
+
+const styleOverrideBottomLeft = css`
+  ${inputContainerStyle}
+  border-radius: 0;
+`;
+
+const styleOverrideBottomRight = css`
+  ${inputContainerStyle}
+  border-radius: 0 4px 4px 0;
+`;
+
+function StackedInputs(props) {
+  const {
+    lockInput,
+    handleChange,
+    inputProps,
+    suffix,
+    firstInputLabel,
+    showLockedSuffixIcon,
+  } = props;
```

Comment:
My idea about this component (or set of components) was a bit different actually.

I'd like to see it be more generic (why specific to four inputs? Why look at those specific labels `topLeft` and `left`? etc) and generalized in one of two ways.

It could either be made as a wrapper with inputs inside like a two-component utility (we use a similar method for `<PillGroup />` and `<Pill />` in the design system):

```js
<StackableGroup>
  <StackableInput
    value=...
    suffix=...
    ...
  />
</StackableGroup>
```

Or it could be achieved using a single component, but with the option of providing an array of inputs.

```js
const inputs = [
  {
    key: 'left',
    value: border.left,
    label: _('Left border', 'web-stories'),
    suffix: <BorderLeft />,
    ...
  },
  ...
];
... 
<StackedInputs
  inputs={inputs}
  onChange=...
  ...
/>
```

I personally prefer the first method, as it seems like a more neat API inside the component using the stacked inputs, but it might be a bit trickier.

The only things that should go inside the new component(s) are basically the look and feel of the inputs and their wrapper and nothing about the logic, naming, etc at all.

In either case, all the stuff about locking/unlocking, the keys, names, and labels of the attributes, etc should all be moved outside the component and it should be possible to reuse this for a completely different set of stacked inputs with a different number of inputs, different labels and names, with and without icons, etc.

For both methods, the difference between locked and unlocked inputs is just about how many child elements to render - either by using conditional rendering (as before) or simply adding more elements to the array in the second instance.